### PR TITLE
sunshine: 0.20.0 -> 0.21.0

### PR DESCRIPTION
**Pull request by devusb**
## Description of changes

https://github.com/LizardByte/Sunshine/releases/tag/v0.21.0

- add new dependencies
- remove now-unnecessary patch
- fix desktop item description

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Melechtna** [comment]
Tested personally, working great on my end!

**Review by Melechtna — APPROVED** [review]
Working for me

**Comment by azuwis** [comment]
Looks like introducing `intel-media-sdk` breaks aarch64-linux https://logs.ofborg.org/?key=nixos/nixpkgs.262591&attempt_id=c3a350c1-d333-43d2-9ba8-80267f87f140.

sunshine 0.20.0 builds fine on aarch64-linux https://hydra.nixos.org/build/238598590

File: pkgs/servers/sunshine/default.nix [hunk 1]
```diff
@@ -109,13 +102,25 @@ stdenv.mkDerivation rec {
...
+    libdatrie
+    xorg.libXdmcp
+    libxkbcommon
+    libepoxy
     libva
     libvdpau
     numactl
     mesa
     amf-headers
+    intel-media-sdk
```
Comment (azuwis) [inline pkgs/servers/sunshine/default.nix:120]:
```suggestion
```

**Review by azuwis — CHANGES_REQUESTED** [review]

**Review by azuwis — APPROVED** [review]

**Pull request merged**